packages/editor/src/components/post-title/index.js

```diff
@@ -7,242 +7,65 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import {
-	forwardRef,
-	useEffect,
-	useImperativeHandle,
-	useRef,
-	useState,
-} from '@wordpress/element';
+import { forwardRef, useState } from '@wordpress/element';
 import { decodeEntities } from '@wordpress/html-entities';
-import { ENTER } from '@wordpress/keycodes';
-import { useSelect, useDispatch } from '@wordpress/data';
-import { pasteHandler } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
 import { store as blockEditorStore } from '@wordpress/block-editor';
-import {
-	__unstableUseRichText as useRichText,
-	create,
-	toHTMLString,
-	insert,
-} from '@wordpress/rich-text';
-import { useMergeRefs } from '@wordpress/compose';
-import { __unstableStripHTML as stripHTML } from '@wordpress/dom';
 
 /**
  * Internal dependencies
  */
-import PostTypeSupportCheck from '../post-type-support-check';
 import { store as editorStore } from '../../store';
-
-/**
- * Constants
- */
-const REGEXP_NEWLINES = /[\r\n]+/g;
+import PostTitleRich from './post-title-rich';
+import { DEFAULT_CLASSNAMES, REGEXP_NEWLINES } from './constants';
+import usePostTitleFocus from './use-post-title-focus';
+import usePostTitle from './use-post-title';
 
 function PostTitle( _, forwardedRef ) {
```

Comment:
So this is basically `PostTitleRich`? It's a bit confusing because this naming implies it's shared, but inside it's only using `PostTitleRich`.

Reply:
Yes we could just everything inside `PostTitle` and remove `PostTitleRich`.

Reply:
I've now removed the `PostTitleRich` component.